javalib/src/main/scala/scala/scalanative/nio/fs/NativePosixFileAttributeView.scala

```diff
@@ -58,10 +64,13 @@ final class NativePosixFileAttributeView(path: Path, options: Array[LinkOption])
 
   override def setGroup(group: GroupPrincipal): Unit =
     Zone { implicit z =>
-      val _group = getGroup(toCString(group.getName))
-      val err    = unistd.chown(toCString(path.toString), -1.toUInt, _group._2)
+      val gid = group match {
+        case g: NativeGroupPrincipal => g.gid
+
+        case _ => throw new IllegalArgumentException("unsupported GroupPrincipal")
```

Comment:
We could also support "normal" principals, with only a name by doing a lookup here.

Reply:
Claudio,
Please let me test my understanding.  Looks to me like GroupPrinciple is at the
bottom of the type hierarchy and that NativeGroupPrinciple is the only
concrete implementer of GroupPrinciple.

What do you mean by "normal". Normal in the type hierarchy or normal as
meant by anyone actually trying to use the method?   Seems to me that the
code you quoted is correct for the former, and that the latter is the concern
of the caller of the method.   Seems like the routine in its JVM defined scope
is correct as is.

Am I off the rails here?  You have been exercising this area and know it.

Reply:
Hi Lee, 

yes, with "normal" I was meaning anyone else calling the method with some sort of custom `GroupPrincipal`.

I thought it should be safe to assume that a caller of the method got the GroupPrincipal by either calling the lookup method of the UserPrincipalLookupService or from other functions of the javalib module which all return a concrete `NativeGroupPrincipal`. (same goes for the UserPrincipal)

Sorry for the confusion!

Reply:
I thought about this a lot and wrote out a long complicated answer describing a
time consuming solution.  

Then it dawned on me that the code in question is part of java.nio and the only
people potentially going to be creating implementations of GroupPrincipal and UserPrincipal
are going to be SN devos.  

Seem to be that the code you quoted is a good sanity check for devos & corruption 
and that we do not need to restrict access to the Principal trait to thwart actively hostile users.